libraries/ESP8266mDNS/src/LEAmDNS2Host_Transfer.cpp

```diff
@@ -0,0 +1,2470 @@
+            break;
+        }
+
+        DEBUG_EX_INFO_IF((bResult) && (p_rpRRAnswer),
+        {
+            DEBUG_OUTPUT.printf_P(PSTR("%s _readRRAnswer: "), _DH());
+            _printRRDomain(p_rpRRAnswer->m_Header.m_Domain);
+            DEBUG_OUTPUT.printf_P(PSTR(" Type:%s Class:0x%04X TTL:%u, RDLength:%u "),
+                                  _RRType2Name(p_rpRRAnswer->m_Header.m_Attributes.m_u16Type),
+                                  (p_rpRRAnswer->m_Header.m_Attributes.m_u16Class | (p_rpRRAnswer->m_bCacheFlush ? 0x8000 : 0)),
+                                  p_rpRRAnswer->m_u32TTL,
+                                  u16RDLength);
+            switch (header.m_Attributes.m_u16Type /*& (~0x8000)*/)      // Topmost bit might carry 'cache flush' flag
+            {
+#ifdef MDNS_IPV4_SUPPORT
+            case DNS_RRTYPE_A:
+                DEBUG_OUTPUT.printf_P(PSTR("A IP:%s"), ((clsRRAnswerA*&)p_rpRRAnswer)->m_IPAddress.toString().c_str());
+                break;
+#endif
+            case DNS_RRTYPE_PTR:
+                DEBUG_OUTPUT.printf_P(PSTR("PTR "));
+                _printRRDomain(((clsRRAnswerPTR*&)p_rpRRAnswer)->m_PTRDomain);
+                break;
+            case DNS_RRTYPE_TXT:
+            {
+                size_t  stTxtLength = ((clsRRAnswerTXT*&)p_rpRRAnswer)->m_Txts.c_strLength();
+                char*   pTxts = new char[stTxtLength];
+                if (pTxts)
+                {
+                    ((clsRRAnswerTXT*&)p_rpRRAnswer)->m_Txts.c_str(pTxts);
+                    DEBUG_OUTPUT.printf_P(PSTR("TXT(%u) %s"), stTxtLength, pTxts);
+                    delete[] pTxts;
+                }
+                break;
+            }
+#ifdef MDNS_IPV6_SUPPORT
+            case DNS_RRTYPE_AAAA:
+                DEBUG_OUTPUT.printf_P(PSTR("AAAA IP:%s"), ((clsRRAnswerAAAA*&)p_rpRRAnswer)->m_IPAddress.toString().c_str());
+                break;
+#endif
+            case DNS_RRTYPE_SRV:
+                DEBUG_OUTPUT.printf_P(PSTR("SRV Port:%u "), ((clsRRAnswerSRV*&)p_rpRRAnswer)->m_u16Port);
+                _printRRDomain(((clsRRAnswerSRV*&)p_rpRRAnswer)->m_SRVDomain);
+                break;
+            /*  case DNS_RRTYPE_NSEC:
+                DEBUG_OUTPUT.printf_P(PSTR("NSEC "));
+                _printRRDomain(((stcRRAnswerNSEC*&)p_rpRRAnswer)->m_NSECDomain);
+                for (uint32_t u=0; u<(((stcRRAnswerNSEC*&)p_rpRRAnswer)->m_pNSECBitmap->m_u16BitmapLength * 8); ++u) {
+                    uint8_t byte = ((stcRRAnswerNSEC*&)p_rpRRAnswer)->m_pNSECBitmap->m_pu8BitmapData[u / 8];
+                    uint8_t flag = 1 << (7 - (u % 8)); // (7 - (0..7)) = 7..0
+                    if (byte & flag) {
+                        DEBUG_OUTPUT.printf_P(PSTR(" %s"), _RRType2Name(u));
+                    }
+                }
+                break;*/
+            default:
+                DEBUG_OUTPUT.printf_P(PSTR("generic "));
+                break;
+            }
```

Comment:
This seems to be debug code, and yet it really messes up readability. Does it make sense to put the switch in its own function and include a call here instead? For a non-debug build, the function would never be called, then it shouldn't be included in the bin.

Reply:
It is indeed a debug code as stated by the beginning of the block:
```
        DEBUG_EX_INFO_IF((bResult) && (p_rpRRAnswer),
        {
```

All this library is written this way.
I added the `XXX_IF()` versions of the macros to make them a little more readable.
It does not make sense to move this debug code to another function only for this block, rather a separate issue to handle all of them.

This code is not compiled in when debug mode is disabled.